registry/cache/warming.go

```diff
@@ -91,10 +91,18 @@ func (w *Warmer) Loop(logger log.Logger, stop <-chan struct{}, wg *sync.WaitGrou
 			w.warm(ctx, logger, im.Name, im.Credentials)
 		} else {
 			select {
+			case <-stop:
+				logger.Log("stopping", "true")
 			case <-refresh:
 				imageCreds = imagesToFetchFunc()
 				backlog = imageCredsToBacklog(imageCreds)
-			default:
+			case name := <-w.Priority:
+				logger.Log("priority", name.String())
+				if creds, ok := imageCreds[name]; ok {
+					w.warm(ctx, logger, name, creds)
+				} else {
+					logger.Log("priority", name.String(), "err", "no creds available")
+				}
```

Comment:
I'd be tempted to lift these six lines and their copies (plus comment) in the earlier `select` into a
```golang
// ...
priorityWarm := func (name string) {
  ...
}
```
before the loop.